gateway/api_loader.go

```diff
@@ -918,12 +937,14 @@ func (gw *Gateway) loadApps(specs []*APISpec) {
 
 	gw.DefaultProxyMux.swap(muxer, gw)
 
-	// Swap in the new register
 	gw.apisMu.Lock()
 
-	// release current specs resources before overwriting map
-	for _, curSpec := range gw.apisByID {
-		curSpec.Release()
+	for _, spec := range specs {
+		for _, curSpec := range gw.apisByID {
+			if spec.APIID == curSpec.APIID && spec.Checksum != curSpec.Checksum {
+				curSpec.Release()
```

Comment:
the tests are reporting goroutine leak with `TestReloadGoroutineLeakWithCircuitBreaker` - most probably since circuit breakers are configured in `MakeSpec` regardless of checksum.
adding a block `else if spec.APIID == curSpec.APIID && spec.Checksum == curSpec.Checksum {
				spec.Release()
			}` would fix this.